samza-core/src/main/java/org/apache/samza/runtime/ApplicationRunnerMain.java

```diff
@@ -50,8 +50,12 @@ public static void main(String[] args) throws Exception {
     ApplicationRunnerCommandLine cmdLine = new ApplicationRunnerCommandLine();
     OptionSet options = cmdLine.parser().parse(args);
     Config orgConfig = cmdLine.loadConfig(options);
-    Config config = Util.rewriteConfig(orgConfig);
     ApplicationRunnerOperation op = cmdLine.getOperation(options);
+    invokeApplicationRunner(orgConfig, op);
+  }
+
+  public static ApplicationRunner invokeApplicationRunner(Config orgConfig, ApplicationRunnerOperation op) {
+    Config config = Util.rewriteConfig(orgConfig);
```

Comment:
Config rewrite shouldn't happen in this method since it's orthogonal to rest of the functionality and is already available as a util.

Reply:
We agreed to keep it the same.